schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java

```diff
@@ -368,6 +351,25 @@ public SchemaMetadataInfo getSchemaMetadata(String schemaName) {
         return schemaMetadataStorable != null ? schemaMetadataStorable.toSchemaMetadataInfo() : null;
     }
 
+
+    @Override
+    public SchemaMetadataInfo updateSchemaMetadata(String schemaName, SchemaMetadata schemaMetadata) {
+        if (!schemaName.equals(schemaMetadata.getName())) {
+            throw new IllegalArgumentException("schemaName must match the name in schemaMetadata");
+        }
+        SchemaMetadataStorable givenSchemaMetadataStorable = new SchemaMetadataStorable();
+        givenSchemaMetadataStorable.setName(schemaName);
+
+        SchemaMetadataStorable schemaMetadataStorable = storageManager.get(givenSchemaMetadataStorable.getStorableKey());
+        if (schemaMetadataStorable != null) {
+            schemaMetadataStorable = SchemaMetadataStorable.updateSchemaMetadata(schemaMetadataStorable, schemaMetadata);
```

Comment:
E.g. We can add that check only after ability to support disable, but there needs an ability to switch. Which withou disable as noted you would break compatibility if going latest to all.